### PR TITLE
JS best practices

### PR DESCRIPTION
**Pull request by Alexr16**
## In this PR:
Add event listener to the checkbox
Update items object's value for completed key upon user actions.
Implement a function for the "Clear all completed" button
Update items object's value for completed key upon user actions.
Store the updated array of items in local storage so the user gets the correct list values after the page reloads.
Removed all hardcoded items from the tasks array.
Create a new JavaScript file for the new functionality.
Implement a function for adding a new task (TODO)
Implement a function for deleting a task
Implement a function for editing task descriptions.
By default new tasks should have the property completed set to false and the property index set to the value of the new array length
Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique
All changes to the To-Do List should be saved in local storage.
Fixed Linter errors

### CODE REVIEW

**Comment by isaacpitwa** [comment]
Hi @Alexr16  👋🏽
Good job so far!
There are some issues that you still need to work on to achieve more on the best Javascript classes and the [DRY KISS and YAGNI](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/clean-code/lesson_DRY_KISS_YAGNI.md)   but you are almost there 🏅
## Highlights  ✔️

  - Clean code and optimal code separation (module generation).
  - Fully Functional project ..bravo 👏🏽👏🏽
  - Professional project documentation
  - Best Javascript Coding practices 
  
 ## Required Changes ♻️

 - You have repeatedly used the code on [line](https://github.com/Alexr16/Todo-list/blob/0970ff7f9ec46c32f309a998a0be56a5240b737f/src/modules/store.js#L16). to improve your DRY, you could write a `save()` method to execute the same line 

 - lines [one](https://github.com/Alexr16/Todo-list/blob/0970ff7f9ec46c32f309a998a0be56a5240b737f/src/index.js#L17) and [Two](https://github.com/Alexr16/Todo-list/blob/0970ff7f9ec46c32f309a998a0be56a5240b737f/src/index.js#L18) could be used with out double document element reading. in order to improve KISS rule
 
almost-there

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @isaacpitwa .

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/store.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  static remove(index) {
+    const list = Store.getList();
+    for (let i = 0; i < list.length; i += 1) {
+      if (list[i].index === index) {
+        list.splice(index - 1, 1);
+      }
+    }
+    localStorage.setItem('task', JSON.stringify(list));
```
Comment (isaacpitwa) [inline src/modules/store.js]: You have repeatedly used this code  to improve your DRY, you could write a `save()` method to execute the same line

File: src/modules/store.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  static updateIndex() {
+    const list = Store.getList();
+    for (let j = 0; j < list.length; j += 1) {
+      list[j].index = j + 1;
+    }
+    localStorage.setItem('task', JSON.stringify(list));
```
Comment (isaacpitwa) [inline src/modules/store.js]: You have repeatedly used this code to improve your DRY, you could write a save() method to execute the same line

File: src/modules/store.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  static updateState(value, task) {
+    const list = Store.getList();
+    list[task.index - 1].state = value;
+    localStorage.setItem('task', JSON.stringify(list));
+  }
+
+  static updateStorage() {
+    const list = Store.getList();
+    Store.updateIndex();
+    localStorage.setItem('task', JSON.stringify(list));
```
Comment (isaacpitwa) [inline src/modules/store.js]: You have repeatedly used this code to improve your DRY, you could write a save() method to execute the same line

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    const list = Store.getList();
+    const index = list.length + 1;
+    const newTask = new Task(description, false, index);
+    List.add(newTask);
+    Store.add(newTask);
+  }
+  input.value = '';
+});
+
+window.addEventListener('load', () => {
```
Comment (isaacpitwa) [inline src/index.js:31]: I believecould be made of less load to the server by using Event trigger `window.onload()` instead of the event-listener used.

**Review by isaacpitwa — CHANGES_REQUESTED** [review]

**Comment by Alexr16** [comment]
- Implemented save() method to set local storage.
- Lines [one](https://github.com/Alexr16/Todo-list/blob/0970ff7f9ec46c32f309a998a0be56a5240b737f/src/index.js#L17) and [Two](https://github.com/Alexr16/Todo-list/blob/0970ff7f9ec46c32f309a998a0be56a5240b737f/src/index.js#L18) restructured to use just one document element reading.

**Review by isaacpitwa — APPROVED** [review]
Hi @Alexr16 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**